lib/serialization/base64UrlSpec.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+import { TypeSpec, createValidationErrorMessage } from "./typeSpec";
+import { SerializationOptions } from "./serializationOptions";
+import { SpecPath } from "./specPath";
+
+/**
+ * A type specification that describes how to validate and serialize a Base64Url encoded ByteArray.
+ */
+const base64UrlSpec: TypeSpec<string, Buffer> = {
```

Comment:
Buffer presents a problem in the browser. Right now we have webpack emit basically a 50kb polyfill for it using the "buffer" package. The closest supertype available in the browser is Uint8Array. This PR might not be the time yet, but let's have a task for solving the Buffer problem.

Reply:
This SO post has some decent suggestions about base64 conversions to/from Uint8Array, and some links to MDN that might provide useful context. https://stackoverflow.com/questions/12710001/how-to-convert-uint8-array-to-base64-encoded-string#12713326

Reply:
I'll definitely need your guys' input when it comes to the browser, because I've never done anything with browsers.
Let's handle Buffers and Streams in seperate PRs.